crates/exex/exex/src/manager.rs

```diff
@@ -241,31 +264,212 @@ impl<Node: Unpin> Stream for ExExNotifications<Node> {
 /// A stream of [`ExExNotification`]s. The stream will only emit notifications for blocks that are
 /// committed or reverted after the given head.
 #[derive(Debug)]
-pub struct ExExNotificationsWithHead<Node> {
-    #[allow(dead_code)]
-    components: Node,
+pub struct ExExNotificationsWithHead<P, E> {
+    node_head: Head,
```

Comment:
does this need to be head, or is NumHash sufficient?

Reply:
same reasoning as https://github.com/paradigmxyz/reth/pull/10787#discussion_r1761042980 im